ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1561,18 +1561,16 @@ private InstallSnapshotReplyProto installSnapshotImpl(InstallSnapshotRequestProt
     }
 
     if (reply != null) {
-      if (request.hasLastRaftConfigurationLogEntryProto() && reply.getResult() ==
-              InstallSnapshotResult.SNAPSHOT_INSTALLED) {
+      if (request.hasLastRaftConfigurationLogEntryProto()) {
         // Set the configuration included in the snapshot
-        LogEntryProto newConfLogEntryProto =
-            request.getLastRaftConfigurationLogEntryProto();
-        LOG.info("{}: set new configuration {} from snapshot", getMemberId(),
-            newConfLogEntryProto);
-
-        state.setRaftConf(newConfLogEntryProto);
-        state.writeRaftConfiguration(newConfLogEntryProto);
-        stateMachine.event().notifyConfigurationChanged(newConfLogEntryProto.getTerm(), newConfLogEntryProto.getIndex(),
-            newConfLogEntryProto.getConfigurationEntry());
+        LogEntryProto newConfLogEntryProto = request.getLastRaftConfigurationLogEntryProto();
+        if (!state.getRaftConf().equals(LogProtoUtils.toRaftConfiguration(newConfLogEntryProto))) {
```

Comment:
If we do not call _state.writeRaftConfiguration(newConfLogEntryProto)_ here, a bootstrapping follower might not write it's configuration to disk at all till the StateMachine applies a log containing a configuration entry. 

_ServerState#writeRaftConfiguration()_ is called only from 2 places - installSnapshot and applyLog. If the conf is persisted on disk some other way also, then we are good. If not, we should check that condition also here before deciding to not update the conf entry on disk.

Reply:
Here the conf entry on disk is only loaded when the server is restarted, as if the server is the first time to start, it has no persisted conf.

    // read configuration from the storage
    Optional.ofNullable(storage.readRaftConfiguration()).ifPresent(this::setRaftConf);
Thus there are two cases:

The init conf is as same as the conf which is about to update. This won't affect at all, as it will still write conf when actually applied a log.
The init conf is different from the conf which is about to update. Here the equivalence check will ensure it will do the write.